### ISSUE TITLE
LINK : fatal error LNK1181: cannot open input file 'stdc++.lib' on Windows

### ISSUE THREAD

**Issue by jvalkeal** [oracle/graal#5901]
**Describe the issue**

`JRT` causing linking failure on windows. Having `-H:+AllowJRTFileSystem` adds `stdc++` in https://github.com/oracle/graal/blob/c4de3aed70358d6c2785fe1f0582420137dbb9fe/substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/jdk/JRTFeature.java#L39 which then throws error during a build:

```
2023-01-30T18:28:51.6624600Z LINK : fatal error LNK1181: cannot open input file 'stdc++.lib'
```

**Steps to reproduce the issue**
1. On windows use flag `-H:+AllowJRTFileSystem`

**Describe GraalVM and your environment:**
 - GraalVM version CE 22.3.0
 - JDK major version: 17
 - OS: Microsoft Windows Server 2022 on github hosted runner

**More details**
Similar to https://github.com/oracle/graal/issues/1876 which was fixed by not using `stdc++` if not a posix system.

Also `JNIRegistrationAwt` looks to have similar `stdc++` issue.

**Comment by oubidar-Abderrahim** [comment]
Hi, Thank you for reaching out about this issue, I tried adding the flag to the native-image build command of a simple HelloWorld on Win10 and managed to build it successfully, do I need something more sophisticated than HelloWorld to reproducer? in that case please share a reproducer.

**Comment by jvalkeal** [comment]
Let me try to create a sample where it fails.

**Comment by jvalkeal** [comment]
@oubidar-Abderrahim Here's a sample project https://github.com/jvalkeal/stdcfail which shows failure with GH actions.

**Comment by oubidar-Abderrahim** [comment]
Tracked internally on GR 44527

**Comment by wirthi** [comment]
The Flag `-H:+AllowJRTFileSystem` is only internal and should not be used. Is there a specific reason why you use it?

**Comment by jvalkeal** [comment]
@wirthi Yeah that is a good question and plot kinda thickens from here. Spring projects https://github.com/spring-projects-experimental/spring-cli and https://github.com/spring-projects-experimental/spring-boot-migrator are starting to experiment how to work in native and those use https://github.com/openrewrite/rewrite.

`openrewrite` uses jdk internals and that's when I noticed use of `-H:+AllowJRTFileSystem` makes things work in native.

Should probably cc @jkschneider if he is interested in this topic.

**Comment by wirthi** [comment]
Sorry, we have not working on that ticket internally. I'll check with the dev team if and what we can do on this.

**Comment by jvalkeal** [comment]
It very much looks like this was fixed in 744bafbd5bd56c49edb00924d64d5b65684d38df. Side effect of #7454.
